$Id: fix_footnote.patch 344 2012-06-29 15:38:56Z tg $

Fix FootNote extension:
* API impropriety
* XHTML/1.0 Transitional validity of output

--- a/dist/mediawiki-extensions-base/usr/share/mediawiki-extensions/base/FootNote/Footnote.php
+++ b/dist/mediawiki-extensions-base/usr/share/mediawiki-extensions/base/FootNote/Footnote.php
@@ -63,8 +63,7 @@ function insert_endnotes( &$parser, &$te
 	
 	$ret = "" ;
 	foreach( $footnoteNotes AS $num => $entry ) {
-		$x = " <a name='footnote{$num}'></a>\n";
-		$x = $x . "<li>$entry <a href='#footnoteback{$num}'>↑</a></li>\n" ;
+		$x = " <li id=\"footnote{$num}\">$entry <a href='#footnoteback{$num}'>↑</a></li>\n" ;
 		$ret .= $x ;
 	}
 	$ret = "<hr /><ol>" . $ret . "</ol>" ;
@@ -78,7 +77,7 @@ function insert_endnotes( &$parser, &$te
 	return true;
 }
 
-function parse_footnote( $text, $params, &$parser ) {
+function parse_footnote( $text, $params, $parser ) {
 	$ret = "" ;
 
 	global $footnoteNotes , $footnoteCount, $footnoteRecursionGuard ;
